### DISCUSSION TITLE
ModelSerializer doesn't enforce model constraints ?

### DISCUSSION THREAD

**Discussion by LiorA1** [encode/django-rest-framework#8290] [Potential Issue] [answered]
Hi,
The Docs state:

```
The ModelSerializer class is the same as a regular Serializer class, except that:
  1. It will automatically generate a set of fields for you, based on the model.
  2. It will automatically generate validators for the serializer, such as unique_together validators.
  3. It includes simple default implementations of .create() and .update().
```

But I noticed another behavior using the code:

```python
class UrlMapper(models.Model):

    ...
    def _generate_rand_str():
        len_of_str = 5
        res_str = "string_larger_then_6_characters"
        return res_str

    short_path = models.CharField(default=_generate_rand_str, max_length=6, unique=True)

class UrlMapperSerializer(serializers.ModelSerializer):
    
    class Meta:
        model = UrlMapper
        fields = ['id', 'url', 'short_path', 'hits']
        read_only_fields = ['id', 'short_path', 'hits']
```

The Unique constraint is enforced, but the max_len doesn't.. (**a string larger then 6 characters can be inserted..**)

Didn't know if the problem is at Django or DRF or the docs, so opened here..

Is it something in the DRF source code ?
or the docs need some refreshing ?

**Comment by lovelydinosaur** [comment, marked as answer]
I wouldn't particularly expect defaults to be validated. 🤷‍♂️
I assume it'll just us whatever `_generate_rand_str` returns there, regardless of any validation constraints.

**Reply by LiorA1** [reply]
I checked and it happens also in Django Forms.
So the default value can ignore some constraints..
Its a bit strange, but 🤷‍♂️
